addons/l10n_mn/migrations/1.1/pre-migrate.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details
+
+
+def migrate(cr, version):
+    ## Tax tags: Change the XMLids of tax tags which have been renamed.
+    tax_tag_changes = [
+        # Old tag            New tag
+        ('vat_report_tag38', 'vat_report_temp'),
+        ('vat_report_tag28', 'vat_report_tag27'),
+        ('vat_report_tag29', 'vat_report_tag28'),
+        ('vat_report_tag33', 'vat_report_tag32'),
+        ('vat_report_tag35', 'vat_report_tag34'),
+        ('vat_report_tag36', 'vat_report_tag35'),
+        ('vat_report_tag37', 'vat_report_tag36'),
+        ('vat_report_tag43', 'vat_report_tag38'),
+        ('vat_report_tag45', 'vat_report_tag40'),
+        ('vat_report_tag52', 'vat_report_tag54'),
+        ('vat_report_tag51', 'vat_report_tag53'),
+        ('vat_report_tag49', 'vat_report_tag51'),
+        ('vat_report_tag48', 'vat_report_tag50'),
+        ('vat_report_tag46', 'vat_report_tag48'),
+        ('vat_report_tag44', 'vat_report_tag46'),
+        ('vat_report_tag42', 'vat_report_tag45'),
+        ('vat_report_tag41', 'vat_report_tag44'),
+        ('vat_report_temp',  'vat_report_tag41'),
+    ]
+
+    queries = ['''
+        UPDATE ir_model_data
+        SET name = %s
+        WHERE name = %s
+            AND module = 'l10n_mn'
+            AND model = 'account.account.tag'
+    '''] * len(tax_tag_changes)
+    params = sum([[new_xmlid, old_xmlid] for old_xmlid, new_xmlid in tax_tag_changes], start=[])
+    cr.execute(';\n'.join(queries), params)
+
+    ## Account tags: all account tags to be deleted should be unset from accounts
+    cr.execute('''
+        DELETE FROM account_account_account_tag aaat
+        USING account_account_tag
+        JOIN ir_model_data
+            ON ir_model_data.module = 'l10n_mn'
+            AND ir_model_data.model = 'account.account.tag'
+            AND ir_model_data.res_id = account_account_tag.id
+        WHERE aaat.account_account_tag_id = account_account_tag.id
+            AND (
+                ir_model_data.name LIKE 'tax_report_tag%'
+                OR ir_model_data.name LIKE 'account_tag_operating_%'
+                OR ir_model_data.name LIKE 'account_tag_investing_%'
+                OR ir_model_data.name LIKE 'account_tag_financing_%'
```

Comment:
`_` is any char for `like` operator.
Raw strings needed for this suggestion.
```suggestion
                ir_model_data.name LIKE 'tax\_report\_tag%'
                OR ir_model_data.name LIKE 'account\_tag\_operating\_%'
                OR ir_model_data.name LIKE 'account\_tag\_investing\_%'
                OR ir_model_data.name LIKE 'account\_tag\_financing\_%'
```

Reply:
Brilliant, thanks so much!